### PR TITLE
Remove makefile backport

### PR DESCRIPTION
**Pull request by pquentin**
Closes #2089 

To do so, I first had to switch from _makefile_refs to _io_refs. Why? It turns out that the socket class needs to track references to the file created with `makefile`, in order to know if the underlying socket can be closed safely.  PyPy 2 uses `_makefile_refs` for this, while PyPy 3 and CPython 3 share the same code and use `_io_refs` instead. Since we no longer support Python 2, we can drop the code specific to PyPy 2 and switch to `_io_refs`. This allows to remove our makefile backport that used `_makefile_refs` for PyPy 2 compatibility.

### CODE REVIEW

**Comment by pquentin** [comment]
https://github.com/urllib3/urllib3/pull/2097/checks?check_run_id=1465401839 is a new kind of flakiness that I've never seen before. Tests are much more flaky during US business hours!

Closing/reopening

**Comment by sethmlarson** [comment]
@pquentin That's new to me too, must have picked a bad address/port combo?

File: setup.py [hunk 1]
```diff
@@ -101,7 +101,6 @@
         "urllib3",
         "urllib3.packages",
         "urllib3.packages.ssl_match_hostname",
-        "urllib3.packages.backports",
```
Comment (sethmlarson) [inline setup.py:104]: Woo! This is a big win!

**Review by sethmlarson — APPROVED** [review]
This change looks great, thanks for this! Goodbye `urllib3.packages.backports` :)

**Comment by sethmlarson** [comment]
Bleh PyPy race condition, @pquentin are you happy enough with this PR for me to force a merge without coverage?

**Comment by pquentin** [comment]
I'd prefer to have tests pass and make sure coverage is still 100%. Closing/reopening again

**Comment by pquentin** [comment]
This time we hit the PyPy SQLite bug. Sigh.

**Comment by sethmlarson** [comment]
Finally passed! Let's merge quick before CI changes its mind.

**Pull request merged**